distribution_circuits_website_sale/models/res_config_settings.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 Coop IT Easy SCRLfs
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from ast import literal_eval
+
+from odoo import api, models, fields
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    def _default_cart_reminder_mail_before_timeframe_end(self):
+        try:
+            return (self.env
+                    .ref('distribution_circuits_sale.'
+                         '.mail_template_sale_cart_timeframe_recovery')
+                    .id)
+        except ValueError:
+            return False
+
+    send_reminder_before_timeframe_end = fields.Boolean(
+        "Send Reminder Before Time Frame End",
+    )
+    cart_reminder_mail_before_timeframe_end = fields.Many2one(
+        'mail.template',
+        string='Abandoned Cart Reminder Before End of Time Frame Email',
+        default=_default_cart_reminder_mail_before_timeframe_end,
+        domain="[('model', '=', 'sale.order')]",
+    )
+    cart_abandoned_before_timeframe_end_delay = fields.Float(
+        "Delay Before End of Time Frame",
+        default=24.0,
+        help="Number of hours before the end of the time frame when a "
+             "reminder is sent",
+    )
+
+    @api.model
+    def get_values(self):
+        res = super(ResConfigSettings, self).get_values()
+        param_mgr = self.env['ir.config_parameter']
+        mail_mgr = self.env['mail.template']
+
+        send_reminder = bool(
+            param_mgr.sudo().get_param(
+                'distribution_circuits_sale'
+                '.send_reminder_before_timeframe_end',
+                default=False
+            )
+        )
+
+        try:
+            cart_reminder_mail = literal_eval(
+                param_mgr.sudo().get_param(
+                    'distribution_circuits_sale'
+                    '.cart_reminder_mail_before_timeframe_end',
+                    default='False')
+            )
+        except ValueError:
+            cart_reminder_mail = False
+        # TODO: is it useful to check if cart_reminder_mail exist
+        # before giving it to browse as it is set to the string 'False'
+        # by default?
+        if (cart_reminder_mail
+                and not mail_mgr.browse(cart_reminder_mail).exists()):
+            cart_reminder_mail = (
+                self._default_cart_reminder_mail_before_timeframe_end()
+            )
+
+        try:
+            cart_delay = float(
+                param_mgr.sudo().get_param(
+                    'distribution_circuits_sale'
+                    '.cart_abandoned_before_timeframe_end_delay',
+                    default='24.0'
+                )
+            )
+        except ValueError:
```

Comment:
why you would have an error here?

Reply:
Because of the float conversion. If the entry in the config is not a string representing a float, the float() will fail raising a ValueError.

Reply:
we suppose that users enter data that are compliant with the type. anyway supposing this you need to put ugly try catch everywhere! Can we just suppose that the user will enter the correct type of data in a system paramater entry and get ride of this try catch?

Reply:
@remytms can you just give at last eye on this please?

Reply:
@houssine78 yes I have commited all the changes you have requested. :) You can merge. :)